moveit_core/robot_model/src/robot_model.cpp

```diff
@@ -1110,11 +1110,7 @@ moveit::core::JointModel* moveit::core::RobotModel::getJointModel(const std::str
 
 const moveit::core::LinkModel* moveit::core::RobotModel::getLinkModel(const std::string& name) const
 {
-  LinkModelMap::const_iterator it = link_model_map_.find(name);
-  if (it != link_model_map_.end())
-    return it->second;
-  logError("Link '%s' not found in model '%s'", name.c_str(), model_name_.c_str());
-  return NULL;
+  return const_cast<RobotModel*>(this)->getLinkModel(name);
```

Comment:
i don't understand - wouldn't this be recursive and not work?

why remove the helpful logError message?

Reply:
This simply relays the call to the otherwise fully identical non-const version of the function.
Removing the constness via `const_cast` ensures that it is not recursive.
The log message is retained from the non-const version.

Reply:
For aesthetic reasons I prefer to cast the other way around (make the non-const function call the const version) so that you don't have to cast const-ness away.

Apart from that this method is perfectly valid.

Reply:
In this case you would have to cast constness away for the result type: `const LinkModel*` -> `LinkModel*`, which then requires two `const_casts`. In my opinion that's even harder to read, although it better matches the semantics. However, this is cosmetics.

Reply:
Ah, that explains why you chose to do it this way around.
I agree that this is cosmetics. No need to get out the eye liner :)

Reply:
+1